client/fs_endpoint.go

```diff
@@ -0,0 +1,921 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	"github.com/hashicorp/nomad/acl"
+	"github.com/hashicorp/nomad/client/allocdir"
+	sframer "github.com/hashicorp/nomad/client/lib/streamframer"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hpcloud/tail/watch"
+	"github.com/ugorji/go/codec"
+)
+
+var (
+	allocIDNotPresentErr = fmt.Errorf("must provide a valid alloc id")
+	pathNotPresentErr    = fmt.Errorf("must provide a file path")
+	taskNotPresentErr    = fmt.Errorf("must provide task name")
+	logTypeNotPresentErr = fmt.Errorf("must provide log type (stdout/stderr)")
+	invalidOrigin        = fmt.Errorf("origin must be start or end")
+)
+
+const (
+	// streamFramesBuffer is the number of stream frames that will be buffered
+	// before back pressure is applied on the stream framer.
+	streamFramesBuffer = 32
```

Comment:
maybe call this numStreamBufferFrames ?

Reply:
The type is named StreamFrame so I think it is little clearer to say this is the buffer count for the streamFrame.